src/References/ReferencesListener.php

```diff
@@ -126,15 +126,12 @@ public function postLoad(EventArgs $eventArgs)
                         $property->setValue(
                             $object,
                             new LazyCollection(
-                                static function () use ($id, &$manager, $class, $identifier) {
-                                    $results = $manager
-                                        ->getRepository($class)
+                                static fn () => new ArrayCollection(
+                                    $manager->getRepository($class)
                                         ->findBy([
                                             $identifier => $id,
-                                        ]);
-
-                                    return new ArrayCollection(is_array($results) ? $results : $results->toArray());
-                                }
+                                        ])
+                                )
```

Comment:
Persistence 4.0 adds the array return type to `ObjectRepository::findBy()`, so the `is_array()` check isn't needed anymore and I've just simplified this down to an arrow function.  Same below.